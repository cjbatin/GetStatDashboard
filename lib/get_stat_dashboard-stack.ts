import { CfnOutput, Construct, Stack, StackProps } from '@aws-cdk/core'
import { NodejsFunction } from 'aws-lambda-nodejs-esbuild'
import { Runtime } from '@aws-cdk/aws-lambda'
import { RetentionDays } from '@aws-cdk/aws-logs'
import { DomainName, HttpApi, HttpMethod } from '@aws-cdk/aws-apigatewayv2'
import { LambdaProxyIntegration } from '@aws-cdk/aws-apigatewayv2-integrations'
import { BuildOptions } from 'esbuild'

const esbuildOptions: BuildOptions = {
  target: 'node12',
}

export class GetStatDashboardStack extends Stack {
  constructor (scope: Construct, id: string, props: StackProps) {
    super(scope, id, props)
    const mainHandler = new NodejsFunction(this, 'MainHandler', {
      runtime: Runtime.NODEJS_12_X,
      rootDir: 'src',
      handler: 'index.main',
      esbuildOptions,
      logRetention: RetentionDays.TWO_WEEKS
    })

    const allSitesHandler = new NodejsFunction(this, 'AllSitesHandler', {
      runtime: Runtime.NODEJS_12_X,
      rootDir: 'src',
      handler: 'allSites.main',
      esbuildOptions,
      logRetention: RetentionDays.TWO_WEEKS
    })

    const allProjectsHandler = new NodejsFunction(this, 'AllProjectsHandler', {
      runtime: Runtime.NODEJS_12_X,
      rootDir: 'src',
      handler: 'allProjects.main',
      esbuildOptions,
      logRetention: RetentionDays.TWO_WEEKS
    })

    const projectHandler = new NodejsFunction(this, 'ProjectHandler', {
      runtime: Runtime.NODEJS_12_X,
      rootDir: 'src',
      handler: 'project.main',
      esbuildOptions,
      logRetention: RetentionDays.TWO_WEEKS
    })

    const tagsHandler = new NodejsFunction(this, 'TagsHandler', {
      runtime: Runtime.NODEJS_12_X,
      rootDir: 'src',
      handler: 'tags.main',
      esbuildOptions,
      logRetention: RetentionDays.TWO_WEEKS
    })

    const api = new HttpApi(this, id);
    api.addRoutes({
      methods: [HttpMethod.GET],
      integration: new LambdaProxyIntegration({ handler: mainHandler }),
      path: '/'
    })

    api.addRoutes({
      methods: [HttpMethod.GET],
      integration: new LambdaProxyIntegration({ handler: allSitesHandler }),
      path: '/{apiKey}/allSites'
    })

    api.addRoutes({
      methods: [HttpMethod.GET],
      integration: new LambdaProxyIntegration({ handler: allProjectsHandler }),
      path: '/{apiKey}/allProjects'
    })

    api.addRoutes({
      methods: [HttpMethod.GET],
      integration: new LambdaProxyIntegration({ handler: projectHandler }),
      path: '/{apiKey}/project/{projectId}'
    })

    api.addRoutes({
      methods: [HttpMethod.GET],
      integration: new LambdaProxyIntegration({ handler: tagsHandler }),
      path: '/{apiKey}/site/{siteId}/tags'
    })

    api.addRoutes({
      methods: [HttpMethod.GET],
      integration: new LambdaProxyIntegration({ handler: allProjectsHandler }),
      path: '/processAPIKey'
    })
  }
}