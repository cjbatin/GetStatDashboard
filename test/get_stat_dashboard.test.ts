import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as GetStatDashboard from '../lib/get_stat_dashboard-stack';

const regions = { IRELAND: 'eu-west-1'}
const env = { account: '07609764445', region: regions.IRELAND }

test('Empty Stack', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new GetStatDashboard.GetStatDashboardStack(app, 'MyTestStack', { env: env });
    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
