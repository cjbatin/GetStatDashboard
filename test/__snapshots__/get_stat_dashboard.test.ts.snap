// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Stack 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258ArtifactHash854E0BFF": Object {
      "Description": "Artifact hash for asset \\"0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258\\"",
      "Type": "String",
    },
    "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3Bucket834DBD8E": Object {
      "Description": "S3 bucket for asset \\"0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258\\"",
      "Type": "String",
    },
    "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1": Object {
      "Description": "S3 key for asset version \\"0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258\\"",
      "Type": "String",
    },
    "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24ArtifactHashBA91B77F": Object {
      "Description": "Artifact hash for asset \\"67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24\\"",
      "Type": "String",
    },
    "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3Bucket4D46ABB5": Object {
      "Description": "S3 bucket for asset \\"67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24\\"",
      "Type": "String",
    },
    "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3VersionKeyB0F28861": Object {
      "Description": "S3 key for asset version \\"67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AllProjectsHandler6F541D1F": Object {
      "DependsOn": Array [
        "AllProjectsHandlerServiceRoleCAEE5B21",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3Bucket834DBD8E",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "allProjects.main",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AllProjectsHandlerServiceRoleCAEE5B21",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "AllProjectsHandlerLogRetentionC9DE5B0A": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "AllProjectsHandler6F541D1F",
              },
            ],
          ],
        },
        "RetentionInDays": 14,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "AllProjectsHandlerServiceRoleCAEE5B21": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "AllSitesHandlerE8329402": Object {
      "DependsOn": Array [
        "AllSitesHandlerServiceRole01BF4BEA",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3Bucket834DBD8E",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "allSites.main",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AllSitesHandlerServiceRole01BF4BEA",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "AllSitesHandlerLogRetentionC4F768AF": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "AllSitesHandlerE8329402",
              },
            ],
          ],
        },
        "RetentionInDays": 14,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "AllSitesHandlerServiceRole01BF4BEA": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3Bucket4D46ABB5",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3VersionKeyB0F28861",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3VersionKeyB0F28861",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "MainHandler0B2856CD": Object {
      "DependsOn": Array [
        "MainHandlerServiceRole74C86798",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3Bucket834DBD8E",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.main",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MainHandlerServiceRole74C86798",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "MainHandlerLogRetentionD3AE5343": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "MainHandler0B2856CD",
              },
            ],
          ],
        },
        "RetentionInDays": 14,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "MainHandlerServiceRole74C86798": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MyTestStackDefaultStage8B49E545": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "AutoDeploy": true,
        "StageName": "$default",
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
    "MyTestStackFB7ADC2B": Object {
      "Properties": Object {
        "Name": "MyTestStack",
        "ProtocolType": "HTTP",
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "MyTestStackGET9991FD25": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "RouteKey": "GET /",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "MyTestStackHttpIntegrationa1137914c48e860ad2a12f9f4219290b403DD5DB",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "MyTestStackGETMyTestStackGET4CD189ADPermission72758973": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "MainHandler0B2856CD",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:076097644456:",
              Object {
                "Ref": "MyTestStackFB7ADC2B",
              },
              "/*/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyTestStackGETapiKeyallProjects32BC2490": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "RouteKey": "GET /{apiKey}/allProjects",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "MyTestStackHttpIntegration09fb62192496fd7b5e86dc6f33ce5ee067A5DF16",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "MyTestStackGETapiKeyallProjectsMyTestStackGETapiKeyallProjects36CD2533Permission46F64907": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "AllProjectsHandler6F541D1F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:076097644456:",
              Object {
                "Ref": "MyTestStackFB7ADC2B",
              },
              "/*/*/{apiKey}/allProjects",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyTestStackGETapiKeyallSitesC6AA0702": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "RouteKey": "GET /{apiKey}/allSites",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "MyTestStackHttpIntegratione30ca213cd1440c806b67f1ebdd7ec977C6D94D6",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "MyTestStackGETapiKeyallSitesMyTestStackGETapiKeyallSitesAB2D967EPermissionB822F6DC": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "AllSitesHandlerE8329402",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:076097644456:",
              Object {
                "Ref": "MyTestStackFB7ADC2B",
              },
              "/*/*/{apiKey}/allSites",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyTestStackGETapiKeysitesiteIdtags3C2BF5E7": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "RouteKey": "GET /{apiKey}/site/{siteId}/tags",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "MyTestStackHttpIntegration49358a76f3eddc26967dc99a83d5b6e588E4FA9C",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "MyTestStackGETapiKeysitesiteIdtagsMyTestStackGETapiKeysitesiteIdtags9A5E5143PermissionE49D2C7B": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "TagsHandler2E9D7B6F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:076097644456:",
              Object {
                "Ref": "MyTestStackFB7ADC2B",
              },
              "/*/*/{apiKey}/site/{siteId}/tags",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyTestStackHttpIntegration09fb62192496fd7b5e86dc6f33ce5ee067A5DF16": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::GetAtt": Array [
            "AllProjectsHandler6F541D1F",
            "Arn",
          ],
        },
        "PayloadFormatVersion": "2.0",
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "MyTestStackHttpIntegration49358a76f3eddc26967dc99a83d5b6e588E4FA9C": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::GetAtt": Array [
            "TagsHandler2E9D7B6F",
            "Arn",
          ],
        },
        "PayloadFormatVersion": "2.0",
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "MyTestStackHttpIntegrationa1137914c48e860ad2a12f9f4219290b403DD5DB": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::GetAtt": Array [
            "MainHandler0B2856CD",
            "Arn",
          ],
        },
        "PayloadFormatVersion": "2.0",
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "MyTestStackHttpIntegratione30ca213cd1440c806b67f1ebdd7ec977C6D94D6": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "MyTestStackFB7ADC2B",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::GetAtt": Array [
            "AllSitesHandlerE8329402",
            "Arn",
          ],
        },
        "PayloadFormatVersion": "2.0",
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "TagsHandler2E9D7B6F": Object {
      "DependsOn": Array [
        "TagsHandlerServiceRoleDCDF5EA9",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3Bucket834DBD8E",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0e9b7f4dc5b6bb412088a16feafd9c294adac1f0351ac0e92d93f0c5a87be258S3VersionKey0C50FAA1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "tags.main",
        "Role": Object {
          "Fn::GetAtt": Array [
            "TagsHandlerServiceRoleDCDF5EA9",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "TagsHandlerLogRetention02056104": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "TagsHandler2E9D7B6F",
              },
            ],
          ],
        },
        "RetentionInDays": 14,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "TagsHandlerServiceRoleDCDF5EA9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
